cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/GoalBasedOptimizationParameters.java

```diff
@@ -52,21 +70,9 @@ public boolean excludeRecentlyRemovedBrokers() {
     return _excludeRecentlyRemovedBrokers;
   }
 
-  public static class GoalsAndRequirements {
-    private final List<String> _goals;
-    private final ModelCompletenessRequirements _requirements;
-
-    public GoalsAndRequirements(List<String> goals, ModelCompletenessRequirements requirements) {
-      _goals = goals; // An empty list indicates the default goals.
-      _requirements = requirements;
-    }
-
-    public List<String> goals() {
-      return _goals;
-    }
-
-    public ModelCompletenessRequirements requirements() {
-      return _requirements;
-    }
+  private static ModelCompletenessRequirements getRequirements(ParameterUtils.DataFrom dataFrom) {
+    return new ModelCompletenessRequirements(MIN_NUM_VALID_WINDOWS.get(dataFrom),
+                                             MIN_VALID_PARTITIONS_RATIO.get(dataFrom),
+                                             INCLUDE_ALL_TOPICS);
```

Comment:
It sounds like an orthogonal proposal to this patch -- I am not sure if we would want to remove functionality, but if you feel like there is room for improvement, would you create an issue to investigate?

-- As a side-note, it would be helpful to provide links to the relevant code line or full name of the `Class#Function` when discussing the implementation details at other parts of the code.

Reply:
sure, I will follow up on this one.